kore/src/Kore/Log/DebugAppliedRewriteRules.hs

```diff
@@ -54,10 +54,8 @@ import Log
 
 data DebugAppliedRewriteRules =
     DebugAppliedRewriteRules
-        { configuration
-            :: !(Pattern Variable)
-        , appliedRewriteRules
-            :: ![UnifiedRule Variable (RewriteRule Variable)]
+        { configuration :: Pattern Variable
```

Comment:
Why did you remove the strictness annotations here and below?

Reply:
The log entry data types shouldn't be strict because we don't want the fields to be evaluated if that entry type is disabled, and if an entry type is enabled, we want evaluation to happen in the logging thread, not in the main thread.

Reply:
Since the standard thing to do is to have strict fields, and this explanation might not be easy to discover for someone who doesn't already know it, would it be worth to add a code comment?